triggers/service/auth/src/main/scala/com/daml/oauth/middleware/Server.scala

```diff
@@ -191,4 +189,39 @@ object Server extends StrictLogging {
         }
     )
   }
+
+  private def refresh(config: Config)(implicit system: ActorSystem, ec: ExecutionContext) =
+    entity(as[Request.Refresh]) { refresh =>
+      val body = OAuthRequest.Refresh(
+        grantType = "refresh_token",
+        refreshToken = refresh.refreshToken,
+        clientId = config.clientId,
+        clientSecret = config.clientSecret)
+      import com.daml.oauth.server.Request.Refresh.marshalRequestEntity
+      val tokenRequest =
+        for {
+          entity <- Marshal(body).to[RequestEntity]
+          req = HttpRequest(uri = config.oauthToken, entity = entity, method = HttpMethods.POST)
```

Comment:
Reading through this, the field name is a bit confusing. `config.endpoints.oauthToken` or `config.oauthTokenEndpoint` might be clearer but let’s leave that out of this PR.

Reply:
Indeed, it might make sense to couple this setting with the configuration of the requests themselves.